app/assets/javascripts/announcements.js

```diff
@@ -1,29 +1,36 @@
-function hideAnnouncement(announcement_key) {
-  createCookie(announcement_key, "hidden", 365);
-  $("#announcement").slideUp();
-}
+$(function() {
 
-// Utility scripts from http://www.quirksmode.org/js/cookies.html
-function createCookie(name, value, days) {
-  if (days) {
-    var date = new Date();
-    date.setTime(date.getTime() + (days * 24 * 60 * 60 * 1000));
-    var expires = "; expires=" + date.toGMTString();
-  } else var expires = "";
-  document.cookie = name + "=" + value + expires + "; path=/";
-}
+  var announcement = $('#announcement');
+  var hideAnnouncement = $('#announcement .hide > a');
 
-function readCookie(name) {
-  var nameEQ = name + "=";
-  var ca = document.cookie.split(';');
-  for (var i = 0; i < ca.length; i++) {
-    var c = ca[i];
-    while (c.charAt(0) == ' ') c = c.substring(1, c.length);
-    if (c.indexOf(nameEQ) == 0) return c.substring(nameEQ.length, c.length);
+  hideAnnouncement.on('click', function() {
+    announcement_key = announcement.data("announcement-token");
+    createCookie(announcement_key, "hidden", 365);
+    $("#announcement").slideUp();
+  });
+
+  // Utility scripts from http://www.quirksmode.org/js/cookies.html
+  function createCookie(name, value, days) {
```

Comment:
Would it make sense to move this utility scripts to its own file called: `cookies.js`?